src/storage.ts

```diff
@@ -265,7 +265,7 @@ const RETRYABLE_ERR_FN_DEFAULT = function (err?: ApiError) {
 
     if (err.errors) {
       for (const e of err.errors) {
-        const reason = e?.reason?.toLowerCase();
+        const reason = e?.reason?.toString().toLowerCase();
```

Comment:
@ddelgrosso1 not sure why `as string` in the gcs-resumable-upload library didn't work, but this should fix it. I think this is a good change to have in our function anyway because it makes our function safer. I will update the resumable library as well, but this should unblock us for now.